### PR TITLE
feat: Convert generated eslintrc file from JSON to CJS

### PR DESCRIPTION
**Pull request by bai**
Closes #<issue>

## ✅ Checklist

- [x] I have followed every step in the [contributing guide](https://github.com/t3-oss/create-t3-app/blob/main/CONTRIBUTING.md) (updated 2022-10-06).
- [x] The PR title follows the convention we established [conventional-commit](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] I performed a functional test on my final commit

---

## Changelog

Rename generated `.eslintrc.json` to `.eslintrc.cjs` and enable type-checking vis JSDoc.

💯

### CODE REVIEW

File: cli/template/base/tsconfig.json [hunk 1]
```diff
@@ -21,6 +21,13 @@
       "~/*": ["./src/*"]
     }
   },
-  "include": ["next-env.d.ts", "**/*.ts", "**/*.tsx", "**/*.cjs", "**/*.mjs"],
+  "include": [
+    ".eslintrc.cjs",
```
Comment (bai) [inline cli/template/base/tsconfig.json:25]: Added to ensure TypeScript checks the file (files in project root are not matched by `**/*.cjs`).

Comment (juliusmarminge) [reply cli/template/base/tsconfig.json:25]: fyi: files in root are matched but files starting in `.` isn't.

anywho this is needed nevertheless

**Review by lodisy — APPROVED** [review]

File: cli/template/base/tsconfig.json (hunk 1 above)
Comment (bai) [reply cli/template/base/tsconfig.json:25]: Ah, TIL!

**Review by juliusmarminge — APPROVED** [review]

**Comment by JohnBakhmat** [comment]
LGTM ;)

**Review by fernandoxu — APPROVED** [review]

**Pull request merged**